src/main/java/hudson/plugins/jira/JiraSite.java

```diff
@@ -254,9 +294,39 @@ protected Object readResolve() {
         projectUpdateLock = new ReentrantLock();
         issueCache = makeIssueCache();
         jiraSession = null;
+
+        // HACK, update final fields via reflection, see jls 17.5.3, https://docs.oracle.com/javase/specs/jls/se7/html/jls-17.html#jls-17.5.3
+        if (StringUtils.isNotBlank(credentialsId)) {
+            StandardUsernamePasswordCredentials credentials = CredentialsHelper.lookupSystemCredentials(credentialsId, url);
+            setCredentials(credentials);
```

Comment:
shouldn't we call `setCredentialsId` here as well?

Reply:
There is a condition that credentialsId is not blank but no matched credentials. ```setCredentialsId``` should be called under non-null check with credentials :(